apps/factory/src/components/prompt_version.tsx

```diff
@@ -212,12 +235,39 @@ function PromptVersion({
     }
   };
 
+  // this function will run during unmounting
+
+  // const cleanup = () => {
+  //   const localStorageKey = getLocalStorageKey(
+  //     lpv.promptPackageId,
+  //     lpv.promptTemplateId,
+  //     lpv.version
+  //   );
+  //   localStorage.removeItem(localStorageKey);
+  // };
+
   const handleChangeProviderModel = () => {
     // logic to change the value of promptInputs and variables
     let newPrompt, newPromptInputs;
+
+    // store prompt instead of promptInputs in localstorage
+    const localStorageKey = getLocalStorageKey(
+      lpv.promptPackageId,
+      lpv.promptTemplateId,
+      lpv.version,
+    );
```

Comment:
This code is called 2 times. you can set the key for a version when the version is created or loaded, it doesnto need to reevaluated. Isnt ?